src/usecases/counter/factories.rs

```diff
@@ -54,11 +54,21 @@ pub async fn create_listening_creator_with_server(
         .map_err(|e| anyhow::anyhow!("Failed to get counter address: {}", e))?;
     let provider = CounterProvider::new(counter_address, provider.clone());
     let queue = SimpleTaskQueue::new();
```

Comment:
move this below `timeout_ms` declaration for better readability. maybe it can be reduced to use less local variables and just create queue inline in `start_http_server()`?